framework/mesh/mesh_generator/distributed_mesh_generator.h

```diff
@@ -0,0 +1,112 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "framework/mesh/mesh_generator/mesh_generator.h"
+
+namespace opensn
+{
+
+/**
+ * @class opensn::DistributedMeshGenerator
+ * @brief Generates and partitions a mesh on location 0, distributing the partitions via MPI.
+ *
+ * This class is responsible for generating a mesh, partitioning it, and distributing the
+ * individual partitions to different MPI locations. The mesh is generated on location 0,
+ * partitioned into multiple parts, serialized, and distributed to all other MPI ranks.
+ *
+ * The mesh data is serialized into a `ByteArray` for efficient MPI communication and
+ * deserialized at the receiving locations to reconstruct the mesh locally.
+ *
+ * @note This class is derived from the MeshGenerator class.
+ */
+class DistributedMeshGenerator : public MeshGenerator
+{
+public:
+  /**
+   * @brief Get the input parameters for this class.
```

Comment:
I don't think we use the brief tag at all...

Reply:
Fixed.